### ISSUE TITLE
Summarization Evaluation on large system output file:

### ISSUE THREAD

**Issue by pfliu-nlp** [neulab/ExplainaBoard#130]
It seem that our web app can not deal with [these files](https://drive.google.com/drive/folders/1TPE2M3R38_ugCAMJ05lEalqErkLhIeBr) 
 (50M) currently?

**Comment by lyuyangh** [comment]
The file size limit is specified [here](https://github.com/neulab/explainaboard_web/blob/main/deployment/nginx.conf). What's the maximum file size we want to support?

**Comment by pfliu-nlp** [comment]
Thank Lyvyang!
>"What's the maximum file size we want to support?"
Good question. I'm not sure whether 100M is too large.
But, as I have mentioned previously in the slack, perhaps enabling the front end to support any submission scenarios is a luxury thing.  The above file is just one system output from one dataset.  In the future, we potentially will have more (multiple systems, multiple languages).
Things we can think about:
* could users only submit the prediction results without the original information of the test set?
  For example, users only need to submit the predicted labels without original text and gold labels in text classification.
* Allow users to submit locally, and they only need to submit the report.json file.
* We could consider other paradigms for system output, which could be reduced to the following generation question:

Given these steps for system outputs, how to design a reasonable lifecycle? And which steps should be conducted by whom and when?
    * acquisition
    * analyzing
    * uploading
    * storing/caching
    * visualization
Specifically,
* which steps should be conducted locally by users or remotely by our server?
* which files should be stored and in what forms (db, disk in server..)
* To what degree do we aim to standardize for each step?


This is one cool paradigm prototype I'm exploring recently: https://github.com/ExpressAI/DataLab/blob/main/tests/operations/infer/test_qc.py 

We also can instantiate other paradigms based on the above context.

**Comment by neubig** [comment]
This is a good question:

> But, as I have mentioned previously in the slack, perhaps enabling the front end to support any submission scenarios is a luxury thing. The above file is just one system output from one dataset. In the future, we potentially will have more (multiple systems, multiple languages).

That's true. If we want this to be someone to use I think handling multiple systems, multiple languages, etc. is something we should probably handle by making our code more efficient and scaling our infrastructure. We want something usable, and don't want to put limits on what people can do unless really necessary.

> could users only submit the prediction results without the original information of the test set? For example, users only need to submit the predicted labels without original text and gold labels in text classification.

This is something I think we should *definitely* have, and in fact I think this is probably the only thing we should support for "named" datasets with training set dependent features like `cnn_dm`. Allowing users to upload their own custom test sets would make the evaluation results not be comparable if the test sets are not exactly the same.

> Allow users to submit locally, and they only need to submit the report.json file.

I'd prefer not to allow this, because (1) this will have the above-stated problems with lack of comparability, (2) this won't allow browsing the system predictions (unless they're included in report.json), and (3) it will require users to install the software with all its dependencies.

> which steps should be conducted locally by users or remotely by our server?

My preference is "all remotely by our server, with some subset of them also being supported by the local SDK".

**Comment by lyuyangh** [comment]
I increased the file size limit to 100M in https://github.com/neulab/explainaboard_web/commit/a40db58f0bb34ad77bd610e961b449771dc38340 (will be deployed in ~10m). I tested with one of the files. The request timed out but the report was generated successfully. This is probably only good for a temporary solution. The file is too big and the processing time is too long for a sync operation. I agree with what Graham said above. And on top of that, I think we probably need to make system submission (report generation) async to solve this issue fundamentally. If supporting large files is a priority, maybe I can focus on this for the next week or so.
